src/pages/TeamDetails/TeamDetails.jsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react'
+import { Link, useLocation, useParams } from "react-router-dom";
+import members from '../../local-json/members.json';
+import bgImg from '../../assets/bg-1.jpg';
+import { Wrapper, Aside, BoxWrap, Box, TopBox, SubTitle } from './TeamDetails.styles';
+import AddMember from '../../utils/AddMember/AddMember';
+import { PieChart } from 'react-minimal-pie-chart';
+
+
+const TeamDetails = () => {
+    const location = useLocation();
+    let { state } = location;
+
+
+    let data = [];
+
+    state.projects.map((obj) => {
+        var randomColor = "#000000".replace(/0/g, function () {
```

Comment:
Generally use map whenever you are going to transform an array into new one or else just use forEach